test/unit/system-state.server.test.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  getIsReconciliationNeeded,
+  initialize,
+  setIsReconciliationNeeded,
+} from '~/models/system-state.server';
+import type { SystemState } from '@prisma/client';
+import { prisma } from '~/db.server';
+
+describe('initialize()', () => {
+  let state: SystemState;
+  beforeAll(async () => {
+    state = await initialize();
+  });
+
+  afterAll(async () => {
+    await prisma.systemState.deleteMany().catch(() => {});
+  });
+
+  test('can create system state', async () => {
+    expect(typeof state).toEqual('object');
+    expect(state.unique).toBe(`unique`);
+    expect(state.reconciliationNeeded).toBe(true);
+  });
+
+  test('only one system state can exist', async () => {
+    await expect(
+      prisma.systemState.create({
+        data: {},
+      })
+    ).rejects.toThrow();
+    await expect(initialize()).rejects.toThrow();
+  });
+});
+
+describe('getIsReconciliationNeeded()', () => {
+  let state: SystemState;
+  beforeAll(async () => {
+    state = await initialize();
+  });
+
+  afterAll(async () => {
+    await prisma.systemState.deleteMany().catch(() => {});
+  });
+
+  test('can get system state correctly', async () => {
+    let result = await getIsReconciliationNeeded();
+    expect(result).toBe(state.reconciliationNeeded);
+  });
+
+  test('can get system state correctly when it does not exist', async () => {
+    await prisma.systemState.deleteMany().catch(() => {});
```

Comment:
An alternative would be to only initialize in the first test, and then delete it there. Not sure if it is any better. What do you think? Same for the next describe block

Reply:
For me, it make sense to start with a default state (system state exist). If anything, I feel I should initialize it again at the end of this test, to restore the default state for other tests that might come after this.

Reply:
Alrighty, makes sense. 

Though if you wanna add more tests to this, I would avoid readding the state. Tests should not depend on each other imo. Better to use a `beforeEach` hook in that case